src/cplus_plugin/gui/constant_raster_manager_dialog.py

```diff
@@ -0,0 +1,1135 @@
+        self.constant_raster_registry.save()
+
+    def _ensure_checked_components_in_collection(self):
+        """Ensure all checked items in the list have components in the collection.
+
+        This method creates components for checked items that don't have them yet,
+        and marks unchecked components as skipped.
+        """
+        # Get current model and collection
+        model = None
+        if self.rb_pathway.isChecked():
+            model = self._pathways_model
+        elif self.rb_activity.isChecked():
+            model = self._activities_model
+
+        if model is None:
+            return
+
+        raster_collection = self.current_constant_raster_collection()
+        if raster_collection is None:
+            return
+
+        # Get the current configuration widget
+        current_config_widget = self.sw_component_container.currentWidget()
+        if not isinstance(current_config_widget, ConstantRasterWidgetInterface):
+            return
+
+        # Get the current value from the widget to use for ALL checked items
+        current_value = None
+        if hasattr(current_config_widget, "sb_experience"):
+            current_value = current_config_widget.sb_experience.value()
+
+        # Track items that were created/updated
+        items_created = []
+        items_updated = []
+
+        # Iterate through all items in the model
+        for row in range(model.rowCount()):
+            item = model.item(row)
+            if item is None:
+                continue
+
+            model_identifier = item.uuid
+            is_checked = item.checkState() == QtCore.Qt.Checked
+
+            # Find or create component for this item
+            component = raster_collection.component_by_id(model_identifier)
+
+            if is_checked:
+                # Item is checked - ensure it has a component
+                if component is None:
+                    # Create a new component
+                    component = current_config_widget.create_raster_component(
+                        item.model_component
+                    )
+                    raster_collection.components.append(component)
+                    items_created.append(item.text())
+                else:
+                    items_updated.append(item.text())
+
+                # Link the model component if not already set
+                if component.component is None:
+                    component.component = item.model_component
+                    component.alias_name = item.text()
+
+                # Apply the current widget value to newly created components
+                # Existing components keep their saved values
+                if items_created and item.text() in items_created:
+                    # This is a new component, apply current widget value
+                    if current_value is not None and component.value_info:
+                        component.value_info.absolute = current_value
+
+                # Mark as enabled
```

Comment:
@osundwajeff enabled is basically for those items that already have a `ConstantRasterComponent` defined in the collection, the `skip_raster` is defined at the collection level though the API provides flexibility to define it at the granular level for edge cases.